llm/data_gemma/dc.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data Commons."""
+
+import concurrent.futures
+import csv
+import io
+from typing import Any, Callable
+
+import requests
+
+from data_gemma import base
+from data_gemma import utils
+
+
+_BASE_URL = 'https://{env}.datacommons.org/nodejs/query'
+
+# Do not allow topics, use higher threshold (0.8).
+_POINT_PARAMS = 'allCharts=1&mode=toolformer_rig&idx=base_uae_mem'
+
+# Allow topics, use lower threshold (0.7)
+_TABLE_PARAMS = 'allCharts=1&mode=toolformer_rag&client=table&idx=base_uae_mem'
+
+
+class DataCommons:
```

Comment:
a more specific name?

Reply:
Do you have a suggestion?

Reply:
The analogy here is :

LLM <-> DataCommons
LLMCall <-> DCCall (maybe DataCommonsCall?)

They both take an input string. LLMCall corresponds to a string, while DCCall corresponds to "data". I feel this class needs to indicate "data" somewhat, rather than giving an answer.

I am really trying to stand as a new user here, though this maybe ok..